_posts/2016-09-29-Rust-1.12.md

```diff
@@ -0,0 +1,305 @@
+1361 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1120-2016-09-29
+
+### What's in 1.12.0 stable
+
+Oh my god! So much stuff!
+
+With 1.12 stable, one of the first visible changes you will notice is the new
+error message format emitted by `rustc`. We've [previously talked] about this
+format and this is the first release where they are broadly available. These
+error messages are a result of the effort of many hours of [volunteer effort] to
+design, test, and update every one of `rustc`s errors to the new format. We're
+excited to see what you think of them:
+
+![A new mismatched trait
+error](/images/2016-09-Rust-1.12/mismatched-trait-error.png)
+
+The other major change in this release is moving to a new compiler backend based
+on the new Rust [MIR]. While this feature does not result in any new
+user-visible features, it paves the way for a number of future compiler
+optimizations, and for some codebases it already results in improvements to
+compile times and reductions in code size.
+
+[previously talked]: https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html
+[volunteer effort]: https://github.com/rust-lang/rust/issues/35233
+[MIR]: https://blog.rust-lang.org/2016/04/19/MIR.html
+
+## Overhauled error messages
+
+With 1.12 we're introducing a new error format which helps to surface a lot of
+the internal knowledge about why an error is occurring to you, the developer. It
+does this by focusing on your code, and the points of interest in your code that
+are relevant to the error.
+
+For example, in 1.11 if a implementation of a trait didn't match the trait
+declaration, you would see an error like the one below:
+
+![An old mismatched trait
+error](/images/2016-09-Rust-1.12/old-mismatched-trait-error.png)
+
+In the new error format we represent the error by instead showing the points in
+the code that matter the most. Here the relevant line in the trait declaration,
+and the relevant line in the implementation, using labels to describe why they
+don't match:
+
+![A new mismatched trait
+error](/images/2016-09-Rust-1.12/mismatched-trait-error.png)
+
+Initially, this error design was built to aid borrow-checking errors, but we
+found, like the error above, the format can be broadly applied to a wide variety
+of errors. If you would like to learn more about the design, check out the
+[previous blog post on the subject][err].
+
+[err]: https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html
+
+## MIR code generation
+
+The new Rust "mid-level IR", usually called "MIR", gives the compiler a simpler
+way to think about Rust code than it's previous way of operating entirely on the
+Rust abstract syntax tree. It makes analysis and optimizations possible that
+have previously been difficult to implement correctly. The first of many
```

Comment:
We just said "previous" in the sentence prior.  Maybe "that would otherwise have been difficult to implement correctly"